Foundation/NSGeometry.swift

```diff
@@ -91,6 +95,61 @@ public func ==(lhs: CGPoint, rhs: CGPoint) -> Bool {
     return lhs.x == rhs.x && lhs.y == rhs.y
 }
 
+extension CGPoint: NSSpecialValueCoding {
+    init(bytes: UnsafePointer<Void>) {
+        let buffer = UnsafePointer<CGFloat>(bytes)
+
+        self.x = buffer.memory
+        self.y = buffer.advancedBy(1).memory
+    }
+    
+    public init?(coder aDecoder: NSCoder) {
+        if aDecoder.allowsKeyedCoding {
+            self = aDecoder.decodePointForKey("NS.pointval")
+        } else {
+            self = aDecoder.decodePoint()
+        }
+    }
+    
+    public func encodeWithCoder(aCoder: NSCoder) {
+        if aCoder.allowsKeyedCoding {
+            aCoder.encodePoint(self, forKey: "NS.pointval")
+        } else {
+            aCoder.encodePoint(self)
+        }
+    }
+    
+    public static func supportsSecureCoding() -> Bool {
+        return true
+    }
+    
+    static func objCType() -> String {
+        return "{CGPoint=dd}"
+    }
+
+    func getValue(value: UnsafeMutablePointer<Void>) {
+        UnsafeMutablePointer<CGPoint>(value).memory = self
+    }
+
+    func isEqual(aValue: Any) -> Bool {
+        if let other = aValue as? CGPoint {
+            return other == self
+        } else {
+            return false
+        }
+    }
+    
+    var hash: Int {
+        return self.x.hash &+ self.y.hash
+    }
+}
+
+extension CGPoint: CustomStringConvertible {
```

Comment:
these methods are internal, there is no way to access them except through NSSpecialValue. I can give them a different name if you think it is confusing with the similarly named NSObject methods.